source/extensions/quic_listeners/quiche/udp_gso_batch_writer.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+// QUICHE allows ignored qualifiers
+#pragma GCC diagnostic ignored "-Wignored-qualifiers"
+
+// QUICHE doesn't mark override at QuicBatchWriterBase::SupportsReleaseTime()
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winconsistent-missing-override"
+#elif defined(__GNUC__) && __GNUC__ >= 5
+#pragma GCC diagnostic ignored "-Wsuggest-override"
+#endif
+
+#include "quiche/quic/core/batch_writer/quic_gso_batch_writer.h"
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#pragma GCC diagnostic pop
+
+#include "envoy/network/udp_packet_writer_handler.h"
+
+#include "common/protobuf/utility.h"
+#include "common/runtime/runtime_protos.h"
+
+namespace Envoy {
+namespace Quic {
+
+#define UDP_GSO_BATCH_WRITER_STATS(COUNTER, GAUGE, HISTOGRAM)                                      \
+  COUNTER(total_bytes_sent)                                                                        \
+  GAUGE(internal_buffer_size, NeverImport)                                                         \
+  GAUGE(front_buffered_pkt_size, NeverImport)                                                      \
```

Comment:
I don't think we need this stats. This is kinda internal state of the writer.

Reply:
Removed this stat. Replaced this with a private state variable `gso_size_`.